src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java

```diff
@@ -1024,4 +1029,47 @@ default void onBlockExploded(BlockState state, World world, BlockPos pos, Explos
         world.setBlockState(pos, Blocks.AIR.getDefaultState(), 3);
         getBlock().onExplosionDestroy(world, pos, explosion);
     }
+
+    /**
+     * Gets the slot that the given state takes up when placed in the same block space as other blocks.<br/>
+     *
+     * Returning {@link MultipartSlot#FULL_BLOCK} signifies that the given state does not support multipart behavior and
+     * is the value returned by default.
+     */
+    default IMultipartSlot getMultipartSlot(BlockState state)
```

Comment:
It would be nice to have an array here. I am thinking about pillar shaped stuff like fences which would block the middle, the top and the bottom, but the other 4 directions (north,west,east,south) would be still awailable.

Reply:
Slots only really exist to reference specific parts of a block and don't actually need to represent the shape of the block they're used on, hence why there is only one and you can register custom ones.

As an example, I might want to place a hollow cover on the top or the bottom of a fence post, but I won't be able to if the fence takes up those slots. It makes sense for a fence to take up the center slot, but not the others, and just occlude its whole shape instead.

Reply:
I can imagien this will run in funny issues when 2 mods register their custom slot which are logic wise in the same space, but I guess there is a case to case basis needed then. 

I already know I have blocks which would block the center and one of the faces, but I guess I just can't register those as multipart then.

Reply:
Would center+occlusion not also work for your blocks?
Or face+occlusion?

Reply:
I never though about that, using center as slot and returing the correct VoxelShape should work perfectly fine because of the occlusion test.

Reply:
From what i understand, the slots are just keys and don't actually matter (much). The actual collision logic is in occlusion.

Reply:
That is correct, and it's what I was suggesting earlier.

Slots are only checked for equality and are purely used as keys to reference "known parts of the block". If there's another part in the same slot, the new part won't be able to be placed.

Occlusion testing is then applied by both parts. If either of them returns a fail, the test fails and the new part can't be added. If either of them returns a success, the test succeeds and the new part can be added. If both of them pass (ignore) the test, which is the default behavior, an intersection test of their occlusion boxes is done in their stead, and the result of that determines whether it can be placed or not.
All in that order, so fail > success > pass.